pkg/querier/blocks_store_queryable.go

```diff
@@ -602,12 +612,30 @@ func (q *blocksStoreQuerier) queryWithConsistencyCheck(
 	return err
 }
 
-func newStoreConsistencyCheckFailedError(remainingBlocks []ulid.ULID) error {
+type storeConsistencyCheckFailedErr struct {
+	remainingBlocks []ulid.ULID
+}
+
+func newStoreConsistencyCheckFailedError(remainingBlocks []ulid.ULID) storeConsistencyCheckFailedErr {
 	// Sort the blocks, so it's easier to test the error strings.
 	sort.Slice(remainingBlocks, func(i, j int) bool {
 		return remainingBlocks[i].Compare(remainingBlocks[j]) < 0
 	})
-	return fmt.Errorf("%v. The failed blocks are: %s", globalerror.StoreConsistencyCheckFailed.Message("failed to fetch some blocks"), strings.Join(convertULIDsToString(remainingBlocks), " "))
+
+	return storeConsistencyCheckFailedErr{
```

Comment:
I'm not clear on why this is now returning the value instead of a pointer? This seems to require the extra check in `Is()` for the pointer and value types.

Reply:
You're right, improved in https://github.com/grafana/mimir/pull/7752/commits/1093b93f6b8c0188b061a80f53ecbca151cb7ddf